examples/stable-diffusion/main.py

```diff
@@ -0,0 +1,122 @@
+import gradio as gr
+import torch
+from torch import autocast
+from diffusers import DiffusionPipeline
+from diffusers import DPMSolverMultistepScheduler
+
+import time
+import hidet
+
+
+model_id = "CompVis/stable-diffusion-v1-4"
+device = "cuda"
+
+dpm = DPMSolverMultistepScheduler.from_pretrained(model_id, subfolder="scheduler")
+pipe = DiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.half, scheduler=dpm).to("cuda")
+
+unet = pipe.unet.eval()
+compiled_unet = torch.compile(unet, backend='hidet')
+
+
+block = gr.Blocks(css=".container { max-width: 800px; margin: auto; }")
+
+num_samples = 1
+eager_time = 0
+compile_time = 0
+
+def infer(prompt, compile):
+    global eager_time, compile_time
+    if compile:
+        pipe.unet = compiled_unet
+    else:
+        pipe.unet = unet
+    
+    with autocast("cuda"):
+        start = time.time()
+        images = pipe([prompt] * num_samples, guidance_scale=7.5).images
+        dur = time.time() - start
+    
+    if compile:
+        compile_time = dur
+    else:
+        eager_time = dur
+    
+    time_text = ""
+    if eager_time == 0:
+        time_text += "eager time: ..."
+    else:
+        time_text += f"eager time: {eager_time}"
+    if compile_time == 0:
+        time_text += " | compile time: ..."
+    else:
+        time_text += f" | compile time: {compile_time}"
+    
+    return images, time_text
+
+def adjust_compile_opts(search_space=0, fp16=True, tensor_core=True, fp_16_red=False, cuda_graph=True, attn=False):
```

Comment:
attn is not used here? Should we give it a shot and see if the pictures are OK?

Reply:
Yes, I fixed a minor issue, but the pictures should be alright, at least with the default settings.

Reply:
Thanks!